### PR TITLE
Integration of lsp-mode diagnostics with dired

### PR DESCRIPTION
**Pull request by yyoncho**
Pretty much all of the code is stolen  from @Alexander-Miller's `treemacs-icons-dired`

### CODE REVIEW

File: lsp-dired.el [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Will add icons for the subdir in the `car' of ARGS."
+  (let* ((path (car args))
+         (pos (cdr (assoc path dired-subdir-alist))))
+    (when pos
+      (lsp-dired--insert-for-subdir path pos))))
+
+(defun lsp-dired--kill-subdir-advice (&rest _args)
+  "Advice to dired kill-subdir commands.
+Will remove the killed subdir from `lsp-dired--covered-subdirs'."
+  (setf lsp-dired--covered-subdirs (delete (dired-current-directory)
```
Comment (nbfalcon) [inline lsp-dired.el:122]: You could use `cl-callf2` here, although it is somewhat obscure:
```emacs-lisp
(cl-callf2 delete (dired-current-directory) lsp-dired--covered-subdirs)
```

**Comment by yyoncho** [comment]
Here it is a screenshot: 

image

**Review by ericdallo — APPROVED** [review]
Really awesome!

**Review by jcs090218 — APPROVED** [review]

File: lsp-dired.el (hunk 1 above)
Comment (yyoncho) [reply lsp-dired.el:121]: To me, it seems like using callf2 will be beneficial when the `place` is more complex while setq/setf seems to be good enough for symbols.

**Review by ericdallo — COMMENTED** [review]
Could you add an entry to CHANGELOG?

**Comment by yyoncho** [comment]
> Could you add an entry to CHANGELOG?

I will update the changelog and the docs in separate PR.

**Pull request merged**